File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/model/Accessible.java
/*
 Copyright Airship and Contributors
 */

package com.urbanairship.android.layout.model;

import com.urbanairship.json.JsonMap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface Accessible {
    @Nullable
    String getContentDescription();

    @Nullable
    static String contentDescriptionFromJson(@NonNull JsonMap json) {
        return json.opt("content_description").getString();
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/model/Accessible.kt
/*
 Copyright Airship and Contributors
 */
package com.urbanairship.android.layout.model

import com.urbanairship.json.JsonMap

internal interface Accessible {

    val contentDescription: String?

    companion object {
        @JvmStatic
        fun contentDescriptionFromJson(json: JsonMap): String? =
            json.opt("content_description").string
    }
}
